client.go

```diff
@@ -624,6 +642,13 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 			}
 		}
 
+		// Check if we should continue with retries.
+		shouldRetry, checkErr = c.CheckRetry(req.Context(), resp, doErr)
+		if !shouldRetry && doErr == nil && req.responseHandler != nil {
+			respErr = req.responseHandler(resp)
```

Comment:
We should log when this returns a non-nil error. Otherwise it wouldn't be clear as to why the client was going to retry. You can see just above here, there's a similar log for when `doErr` is non-nil.